packages/sdks/src/blocks/button/button.lite.tsx

```diff
@@ -39,7 +29,22 @@ export default function Button(props: ButtonProps) {
               role: 'link',
             }
           : { role: 'button' }),
-      }}
+      };
+    },
+  });
+  return (
+    <DynamicRenderer
+      TagName={useTarget({
+        reactNative: props.link
+          ? // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            props.builderLinkComponent || BaseText
+          : // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            BaseText,
+        default: props.link ? props.builderLinkComponent || 'a' : 'button',
+      })}
+      attributes={state.attrs}
```

Comment:
How come this needed to be moved into its own property? I assume something about Angular not supporting spreads? is there any workaround that avoids having a new piece of state?


 I try to reduce the number of things stored in state because depending on the framework, they might not update/cause re-renders when the original prop changes. So this could cause issues in certain SDKs.

Reply:
Unfortunately angular doesn't support calling functions like this with [fn()] and also object spread with ternary doesn't work inside angular template

https://github.com/sidmohanty11/builder/blob/main/packages/sdks/src/blocks/button/button.lite.tsx#L32-L41

Reply:
cc @shyam-builder is there any other workaround?